### PR TITLE
Orbital phase helper method

### PR DESCRIPTION
**Pull request by scottransom**
This fixes all of #475, #598, #686, and #773 (I think).

You use it like:

```
In [1]: import pint.models as model                                                                                                                      

In [2]: m = model.get_model("/home/sransom/timing/0737/0737A_latest.par")                                                                                
INFO: Parameter PBDOT's value will be scaled by 1e-12 [pint.models.parameter]
INFO: Parameter PBDOT's value will be scaled by 1e-12 [pint.models.parameter]
WARNING: Unrecognized parfile line 'NPRNT                    3' [pint.models.timing_model]
WARNING: POSEPOCH not found; using PEPOCH unless set explicitly! [pint.models.astrometry]

In [3]: m.is_binary  # this is also a new method                                                                                                                                     
Out[3]: True

In [4]: m.orbital_phase([56789.234, 56790.765], anom='mean')                                                                                             
Out[4]: <Quantity [1.34467392, 1.17425508] rad>

In [5]: m.orbital_phase([56789.234, 56790.765], anom='eccen')                                                                                            
Out[5]: <Quantity [1.43160206, 1.25776666] rad>

In [6]: m.orbital_phase([56789.234, 56790.765], anom='true')                                                                                             
Out[6]: <Quantity [1.51917692, 1.34253751] rad>

In [7]: m.orbital_phase(56789.234)                                                                                                                       
Out[7]: <Quantity [1.34467392] rad>
```

If you call it with a TOAs object for the times, it will barycenter them first (as it should).

Comments and suggestions welcome.

### CODE REVIEW

**Comment by paulray** [comment]
Wow, awesome! Can it also be called with a Time() object? If so, must it have scale='tdb'?

Also, if I want orbit phase from 0-1, what is the best way?  Is it to use op.to(u.cycle)? Or op.value/(2*np.pi)?

**Comment by scottransom** [comment]
The answer to your first question is tricky.  It should work with Time objects (although right now there is some units bug that is causing that not to work properly), but "tdb" isn't enough.  We really need the times to be *barycentric*, not just tdb.  And there is no good way to force that.  So I'm tempted to just use the Time objects as .value no matter how they are passed and *assume* that that is what the user is giving us.

As for getting 0-1 phases, I have been dividing by (2*np.pi*u.rad).  But I think you can do it in multiple ways.  Maybe that could be a keyword argument to get unitless "full-cycle" phases if requested?

**Comment by paulray** [comment]
> It should work with Time objects (although right now there is some units bug that is causing that not to work properly), but "tdb" isn't enough.

Very good point!  But if a Time object is passed, at a minimum it should be in the 'tdb' scale, otherwise it should throw an exception. And of course the docstring should be very clear that all times passed (unless they are full TOAs) must be barycenter-corrected and in the TDB timescale.

File: src/pint/models/timing_model.py [hunk 1]
```diff
@@ -304,6 +304,40 @@ def phase_funcs(self):
             pfs += p.phase_funcs_component
         return pfs
 
+    @property
+    def is_binary(self):
+        """Does the model describe a binary pulsar? (True or False)"""
+        return any(x.startswith("Binary") for x in self.components.keys())
+
+    def orbital_phase(self, barytimes, anom="mean"):
+        """Return orbital phase (in radians) at barycentric MJD times"""
```
Comment (paulray) [inline src/pint/models/timing_model.py:313]: This docstring needs to be expanded. Should define the possible choices for `anom` and should use the standard docstring style defining Parameters and Returns.

**Comment by scottransom** [comment]
OK.  I've made all the changes that we discussed today:
 - Good docstrings
 - Much improved time handling (including tdb checks and the astropy.Time bug fix)
 - An optional unitless cyclic return (phases in range 0-1)
 - And a .conjunction method which will find the nearest conjunction after an input MJD

I would like to have the conjunctions as an option to plot in pintk, but I'm not sure we want to wait on that for this PR

Any other comments?

**Pull request merged**